packages/sdk/src/index.ts

```diff
@@ -99,21 +83,75 @@ export class WHIPClient {
     this.log("IceCandidateError", e);
   }
 
-  async ingest(mediaStream: MediaStream): Promise<void> {
-    if (this.opts.iceConfigFromEndpoint) {
-      const iceServers: WHIPClientIceServer[] = await this.doFetchICEFromEndpoint();
-      this.peer.setConfiguration({ iceServers: iceServers });
+  private onIceGatheringTimeout() {
+    this.log("IceGatheringTimeout");
+    clearTimeout(this.iceGatheringTimeout);
+
+    this.peer.removeEventListener("icecandidate", this.onIceCandidate.bind(this));
```

Comment:
`.bind(this)` returns a new function so you're not actually removing the listener you added below, you need to store the initial `.bind(this)` reference somewhere or just set and then remove `this.peer.onicecandidate`

Reply:
Aha, thanks!